c++/src/OrcHdfsFile.cc

```diff
@@ -34,15 +34,13 @@
 #include "common/hdfs_configuration.h"
 #include "common/configuration_loader.h"
 
-#define BUF_SIZE 1048576 //1 MB
-
 namespace orc {
 
   class HdfsFileInputStream : public InputStream {
```

Comment:
For a mock that does real network stuff libhdfs++ uses the C++/JVM bindings over the minidfscluster (java namenode and datanode in a single process) for its unit tests.  That seems like a very large dependency to bring into this project.  It'd be possible to write a mock that goes over TCP in C/C++, I can think of 2 ways:
1) Capture network traffic of a read with tcpdump/wireshark for a given request and send the response back.  This would be really brittle if IO request sizes changed and pretty ugly to store a bunch of binary data.
2) Implement at least some of the server side HDFS RPC protocol and DataTransferProtocol to make a mock namenode and datanode.  This is possible but would be a considerable amount of work that wouldn't be as useful for libhdfs++ since it already has the minidfscluster.